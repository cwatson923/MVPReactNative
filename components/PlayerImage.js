import React from "react";
import { View, Text, Image, StyleSheet } from "react-native";
import ResponsiveSize from "../config/getScreenDimensions";

const defaultPlayerImage = require("../assets/icons/Default-user.png");

function PlayerImage (playerData) {
	if (playerData.fullName) {
		const nameData = playerData.fullName.toString().toUpperCase().split(" ");
		const initials = nameData.map((word) => word.charAt(0)).join("");

		return (
			<View style={playerData.large ? (playerData.isWinner ? styles.initialsContainerWinnerLarge : styles.initialsContainerLarge) : (playerData.isWinner ? styles.initialsContainerWinner : styles.initialsContainer)}>
				<Text style={playerData.large ? styles.initialsTextLarge : styles.initialsText}>{initials.substr(0, 2)}</Text>
			</View>
		);
	}

	return (
		<Image style={playerData.isWinner ? styles.pictureWinner : styles.picture} source={defaultPlayerImage} />
	);
}

const styles = StyleSheet.create({
	initialsContainer: {
		height: ResponsiveSize(5),
		width: ResponsiveSize(5),
		borderWidth: 3,
		borderColor: "#BEBEBB",
		borderRadius: 100,
		marginBottom: 5
	},
	initialsContainerLarge: {
		height: 130,
		width: 130,
		borderWidth: 7,
		borderColor: "#BEBEBB",
		borderRadius: 100
	},
	initialsContainerWinner: {
		height: 75,
		width: 75,
		borderWidth: 3,
		borderColor: "#399D60",
		borderRadius: 100,
		marginBottom: 5
	},
	initialsContainerWinnerLarge: {
		height: 130,
		width: 130,
		borderWidth: 7,
		borderColor: "#399D60",
		borderRadius: 100
	},
	initialsText: {
		fontFamily: "KlinicSlab-Book",
		fontSize: ResponsiveSize(10.4),
		fontWeight: "bold",
		letterSpacing: -0.63,
		height: 75,
		color: "#BEBEBB",
		textAlign: "center",
		marginHorizontal: -3,
		marginVertical: 17
	},
	initialsTextLarge: {
		fontFamily: "KlinicSlab-Book",
		fontSize: 72,
		fontWeight: "bold",
		letterSpacing: -0.63,
		width: 130,
		height: 130,
		color: "#BEBEBB",
		textAlign: "center",
		marginHorizontal: -7,
		marginVertical: 30
	},
	picture: {
		height: 75,
		width: 75,
		resizeMode: "contain"
	},
	pictureWinner: {
		height: 75,
		width: 75,
		borderColor: "#399D60",
		borderWidth: 5,
		borderRadius: 100
	}
});

export default PlayerImage;
