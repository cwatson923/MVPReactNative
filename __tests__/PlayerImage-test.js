/**
 * @format
 */

import React from "react";
import "react-native";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import PlayerImage from "../components/PlayerImage";
import ResponsiveSize from "../config/getScreenDimensions";

it("renders correctly", () => {
	renderer.create(<PlayerImage />);
});

it("snapshot renders correctly", () => {
	const tree = renderer.create(<PlayerImage />).toJSON();
	expect(tree).toMatchSnapshot();
	expect(tree.props.source.testUri).toEqual("../../../assets/icons/Default-user.png");
});

it("renders correctly with data - default", () => {
	const tree = renderer.create(<PlayerImage profilePhoto="foo" fullName="bar" />).toJSON();
	expect(tree).toMatchSnapshot();
	expect(tree.props.source.uri).toEqual("foo");
	expect(tree.props.style[0].height).toEqual(ResponsiveSize(5));
	expect(tree.props.style[0].width).toEqual(ResponsiveSize(5));
	expect(tree.props.style[0].borderWidth).toEqual(3);
});

it("renders correctly with data - large", () => {
	const tree = renderer.create(<PlayerImage profilePhoto="foo" fullName="bar" large />).toJSON();
	expect(tree).toMatchSnapshot();
	expect(tree.props.style[0].height).toEqual(150);
	expect(tree.props.style[0].width).toEqual(150);
	expect(tree.props.style[1].height).toEqual(130);
	expect(tree.props.style[1].width).toEqual(130);
	expect(tree.props.style[1].borderWidth).toEqual(7);
});

it("renders correctly with data - winner", () => {
	const tree = renderer.create(<PlayerImage profilePhoto="foo" fullName="bar" isWinner />).toJSON();
	expect(tree).toMatchSnapshot();
	expect(tree.props.style[0].height).toEqual(ResponsiveSize(5));
	expect(tree.props.style[0].width).toEqual(ResponsiveSize(5));
	expect(tree.props.style[0].borderWidth).toEqual(3);
	expect(tree.props.style[2].borderColor).toEqual("#399D60");
});

it("renders correctly with data - winner large", () => {
	const tree = renderer.create(<PlayerImage profilePhoto="foo" fullName="bar" isWinner large />).toJSON();
	expect(tree).toMatchSnapshot();
	expect(tree.props.style[0].height).toEqual(150);
	expect(tree.props.style[0].width).toEqual(150);
	expect(tree.props.style[1].height).toEqual(130);
	expect(tree.props.style[1].width).toEqual(130);
	expect(tree.props.style[1].borderWidth).toEqual(7);
	expect(tree.props.style[2].borderColor).toEqual("#399D60");
});
