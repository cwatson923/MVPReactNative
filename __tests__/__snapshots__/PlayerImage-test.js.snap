// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with data - default 1`] = `
<Image
  source={
    Object {
      "uri": "foo",
    }
  }
  style={
    Array [
      Object {
        "borderColor": "#BEBEBB",
        "borderRadius": 100,
        "borderWidth": 3,
        "height": 150,
        "marginBottom": 5,
        "width": 150,
      },
      Object {},
      Object {},
    ]
  }
/>
`;

exports[`renders correctly with data - initials 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#BEBEBB",
        "borderRadius": 100,
        "borderWidth": 3,
        "height": 150,
        "marginBottom": 5,
        "width": 150,
      },
      Object {},
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#BEBEBB",
          "fontFamily": "KlinicSlab-Bold",
          "fontSize": 72.11538461538461,
          "height": 75,
          "letterSpacing": -0.63,
          "marginHorizontal": -3,
          "marginVertical": 17,
          "textAlign": "center",
        },
        Object {},
      ]
    }
  >
    FB
  </Text>
</View>
`;

exports[`renders correctly with data - initials large 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#BEBEBB",
        "borderRadius": 100,
        "borderWidth": 3,
        "height": 150,
        "marginBottom": 5,
        "width": 150,
      },
      Object {
        "borderRadius": 100,
        "borderWidth": 7,
        "height": 130,
        "width": 130,
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#BEBEBB",
          "fontFamily": "KlinicSlab-Bold",
          "fontSize": 72.11538461538461,
          "height": 75,
          "letterSpacing": -0.63,
          "marginHorizontal": -3,
          "marginVertical": 17,
          "textAlign": "center",
        },
        Object {
          "fontSize": 72,
          "height": 130,
          "marginHorizontal": -7,
          "marginVertical": 22,
          "width": 130,
        },
      ]
    }
  >
    FB
  </Text>
</View>
`;

exports[`renders correctly with data - initials winner 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#BEBEBB",
        "borderRadius": 100,
        "borderWidth": 3,
        "height": 150,
        "marginBottom": 5,
        "width": 150,
      },
      Object {},
      Object {
        "borderColor": "#399D60",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#BEBEBB",
          "fontFamily": "KlinicSlab-Bold",
          "fontSize": 72.11538461538461,
          "height": 75,
          "letterSpacing": -0.63,
          "marginHorizontal": -3,
          "marginVertical": 17,
          "textAlign": "center",
        },
        Object {},
      ]
    }
  >
    FB
  </Text>
</View>
`;

exports[`renders correctly with data - initials winner large 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#BEBEBB",
        "borderRadius": 100,
        "borderWidth": 3,
        "height": 150,
        "marginBottom": 5,
        "width": 150,
      },
      Object {
        "borderRadius": 100,
        "borderWidth": 7,
        "height": 130,
        "width": 130,
      },
      Object {
        "borderColor": "#399D60",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#BEBEBB",
          "fontFamily": "KlinicSlab-Bold",
          "fontSize": 72.11538461538461,
          "height": 75,
          "letterSpacing": -0.63,
          "marginHorizontal": -3,
          "marginVertical": 17,
          "textAlign": "center",
        },
        Object {
          "fontSize": 72,
          "height": 130,
          "marginHorizontal": -7,
          "marginVertical": 22,
          "width": 130,
        },
      ]
    }
  >
    FB
  </Text>
</View>
`;

exports[`renders correctly with data - large 1`] = `
<Image
  source={
    Object {
      "uri": "foo",
    }
  }
  style={
    Array [
      Object {
        "borderColor": "#BEBEBB",
        "borderRadius": 100,
        "borderWidth": 3,
        "height": 150,
        "marginBottom": 5,
        "width": 150,
      },
      Object {
        "borderRadius": 100,
        "borderWidth": 7,
        "height": 130,
        "width": 130,
      },
      Object {},
    ]
  }
/>
`;

exports[`renders correctly with data - winner 1`] = `
<Image
  source={
    Object {
      "uri": "foo",
    }
  }
  style={
    Array [
      Object {
        "borderColor": "#BEBEBB",
        "borderRadius": 100,
        "borderWidth": 3,
        "height": 150,
        "marginBottom": 5,
        "width": 150,
      },
      Object {},
      Object {
        "borderColor": "#399D60",
      },
    ]
  }
/>
`;

exports[`renders correctly with data - winner large 1`] = `
<Image
  source={
    Object {
      "uri": "foo",
    }
  }
  style={
    Array [
      Object {
        "borderColor": "#BEBEBB",
        "borderRadius": 100,
        "borderWidth": 3,
        "height": 150,
        "marginBottom": 5,
        "width": 150,
      },
      Object {
        "borderRadius": 100,
        "borderWidth": 7,
        "height": 130,
        "width": 130,
      },
      Object {
        "borderColor": "#399D60",
      },
    ]
  }
/>
`;

exports[`snapshot renders correctly 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../assets/icons/Default-user.png",
    }
  }
  style={
    Array [
      Object {
        "height": 75,
        "resizeMode": "contain",
        "width": 75,
      },
      Object {},
    ]
  }
/>
`;
